Modules/Tokens/src/Tokens.Domain/Entities/Token.cs

```diff
@@ -38,24 +39,23 @@ public Token(IdentityAddress createdBy, DeviceId createdByDevice, byte[] content
     public IdentityAddress CreatedBy { get; set; }
     public DeviceId CreatedByDevice { get; set; }
 
+    public IdentityAddress? ForIdentity { get; set; }
+
     public byte[] Content { get; private set; }
     public DateTime CreatedAt { get; set; }
     public DateTime ExpiresAt { get; set; }
 
-    public void LoadContent(byte[] content)
-    {
-        if (Content != null)
-            throw new Exception("Cannot change the content of a token.");
-
-        Content = content;
-    }
-
     public static Expression<Func<Token, bool>> IsExpired =>
         challenge => challenge.ExpiresAt <= SystemTime.UtcNow;
 
     public static Expression<Func<Token, bool>> IsNotExpired =>
         challenge => challenge.ExpiresAt > SystemTime.UtcNow;
 
+    public static Expression<Func<Token, bool>> CanBeCollectedBy(IdentityAddress? address)
+    {
+        return token => token.ForIdentity == null || address != null && (token.ForIdentity == address || token.CreatedBy == address);
```

Comment:
Hmm. Do we actually need the `address != null` part? 🤔 

```suggestion
        return token => token.ForIdentity == null || address != null && (token.ForIdentity == address || token.CreatedBy == address);
```

Reply:
changes made